src/Core/Services/Fido2AuthenticatorService.cs

```diff
@@ -125,26 +139,30 @@ public async Task<Fido2AuthenticatorGetAssertionResult> GetAssertionAsync(Fido2A
             var userVerified = response.UserVerified;
 
             var selectedCipher = cipherOptions.FirstOrDefault((c) => c.Id == selectedCipherId);
-            if (selectedCipher == null) {
+            if (selectedCipher == null)
+            {
                 throw new NotAllowedError();
             }
 
-            if (!userVerified && (assertionParams.RequireUserVerification || selectedCipher.Reprompt != CipherRepromptType.None)) {
+            if (!userVerified && (assertionParams.RequireUserVerification || selectedCipher.Reprompt != CipherRepromptType.None))
+            {
                 throw new NotAllowedError();
             }
-            
-            try {
+
+            try
+            {
                 var selectedFido2Credential = selectedCipher.Login.MainFido2Credential;
                 var selectedCredentialId = selectedFido2Credential.CredentialId;
 
-                if (selectedFido2Credential.CounterValue != 0) {
+                await _cipherService.UpdateLastUsedDateAsync(selectedCipher.Id);
+
+                if (selectedFido2Credential.CounterValue != 0)
+                {
                     ++selectedFido2Credential.CounterValue;
+                    var encrypted = await _cipherService.EncryptAsync(selectedCipher);
+                    await _cipherService.SaveWithServerAsync(encrypted);
                 }
```

Comment:
This is the main change, sorry about the formatter generating lots of changed rows